### ISSUE TITLE
invalid memory read in SwapDWord

### ISSUE THREAD

**Issue by bestshow** [Yeraze/ytnef#49]
On Yerase TNEF Printer v1.9.2, an invalid memory read was found in the function SwapDWord.

```
#ytnefprint $FILE
=================================================================
 SEGV on unknown address 0x000000000008 (pc 0x00000052223d bp 0x7ffcf7d97890 sp 0x7ffcf7d976c0 T0)
==16379==The signal is caused by a READ memory access.
==16379==Hint: address points to the zero page.
    #0 0x52223c in SwapDWord /home/haojun/Downloads/ytnef-master/lib/ytnef.c:180:26
    #1 0x52223c in IsCompressedRTF /home/haojun/Downloads/ytnef-master/lib/ytnef.c:1479
    #2 0x52223c in MAPIPrint /home/haojun/Downloads/ytnef-master/lib/ytnef.c:1413
    #3 0x5164c2 in PrintTNEF /home/haojun/Downloads/ytnef-master/ytnefprint/main.c:169:5
    #4 0x51554a in main /home/haojun/Downloads/ytnef-master/ytnefprint/main.c:84:5
    #5 0x7f3f435d4b34 in __libc_start_main /usr/src/debug/glibc-2.17-c758a686/csu/../csu/libc-start.c:274
    #6 0x41a8db in _start (/home/haojun/Downloads/ytnef-afl-build/bin/ytnefprint+0x41a8db)

 SEGV /home/haojun/Downloads/ytnef-master/lib/ytnef.c:180:26 in SwapDWord
==16379==ABORTING
```

testcase: https://github.com/bestshow/p0cs/blob/master/invalid-memory-read-in_SwapDWord
Credit : ADLab of Venustech

**Comment by fgeek** [comment]
Please use CVE-2017-12142 for this issue.

**Comment by ohwgiles** [comment]
It's not valid to call `PrintTNEF` on a `TNEFStruct` that did not correctly pass `TNEFParse`. Issue fixed with 35dc50190

**Comment by Yeraze** [comment]
Released in https://github.com/Yeraze/ytnef/releases/tag/v1.9.3
